src/classes/UTIL_AbstractCommon_BATCH.cls

```diff
@@ -0,0 +1,453 @@
+        if (startId == null) {
+            this.startId = determineStartId();
+            this.endId = determineEndId();
+            this.ldvMode = (ldvMode != true) ? isLDVCircumstance(this.startId, this.endId) : ldvMode;
+        }
+
+        Database.QueryLocator ql = null;
+        Database.QueryLocatorIterator qli = null;
+
+        // If startId is null, there were no records present in the object
+        if (startId != null) {
+            ql = Database.getQueryLocator(getQuery(startId));
+            qli = (ql != null) ? ql.iterator() : null;
+        }
+
+        // If no records were present/returned, it will execute a limited query with a LIMIT 0 in order to avoid
+        // Iterable errors
+        if ((qli == null) || (!qli.hasNext())) {
+            ql = Database.getQueryLocator(getQueryLimitZero());
+            this.endReached = true;
+        }
+
+        return ql;
+    }
+
+    /**
+     * @description Query to be used if object has no records to prevent not iterable error
+     * @return String queryString to execute for Object, limited to 0 records
+     */
+    @testVisible private String getQueryLimitZero() {
+        return 'SELECT Id FROM ' + getSOQLObjectName() + ' LIMIT 0';
+    }
+
+    /**
+     * @description Determines the query to execute factoring all the different params for where clause, ldv versus non ldv, etc.
+     * @param startId starting id of the chunk
+     * @return String queryString to execute for Batch
+     */
+    @testVisible private String getQuery(Id startId) {
+        return 'SELECT ' + getQueryFieldList() + ' FROM ' + getSOQLObjectName() + getQueryWhereClause() + getQueryOrderByAndLimitClause();
+    }
+
+    /**
+     * @description SOQL Query Fragment: Part of the query between the SELECT and FROM, returning either the ID if in LDV
+     * Chunking mode or Subquery is present, or fields to return for the batch Query
+     * @return String part of query between the SELECT and the outer FROM
+     */
+    @testVisible private String getQueryFieldList() {
+        return this.ldvMode || isSubQueryPresent() ? 'Id' : getSOQLFullQueryFieldList();
+    }
+
+    /**
+     * @description SOQL Query Fragment: ORDER BY and LIMIT CLAUSE.  If LDV/Chunking, it will return a default order by ID
+     * with a limit of the Chunk Size for the Batch so the query does not time out
+     * @return String SQOL Query Fragment for the ORDER BY and LIMIT fragment if needed
+     */
+    @testVisible private String getQueryOrderByANDLimitClause() {
+        return this.ldvMode ? ' ORDER BY Id ASC LIMIT ' + getValidLDVChunkSize() : '';
+    }
+
+    /**
+     * @description SOQL Query Fragment: Calculates the where clause for the batch driver in LDV Chunking and non LDV mode
+     * @return String Where clause fragment
+     */
+    @testVisible private String getQueryWhereClause() {
+        String ldvWhere = isFirstChunk() ? ' WHERE Id >= \'' + startId + '\'' : ' WHERE Id > \'' + startId + '\'';
```

Comment:
The ternary conditions are getting a little hard to read here and in many places below. I recommend breaking these up into smaller individual var assignments and then concatenating the results so it's easier to read.